query-service-impl/src/main/java/org/hypertrace/core/query/service/prometheus/FilterToPromqlConverter.java

```diff
@@ -0,0 +1,115 @@
+package org.hypertrace.core.query.service.prometheus;
+
+import static org.hypertrace.core.query.service.QueryRequestUtil.getLogicalColumnNameForSimpleColumnExpression;
+
+import com.google.protobuf.ByteString;
+import java.util.List;
+import java.util.function.Function;
+import org.apache.commons.codec.binary.Hex;
+import org.hypertrace.core.query.service.QueryRequestUtil;
+import org.hypertrace.core.query.service.api.Expression;
+import org.hypertrace.core.query.service.api.Filter;
+import org.hypertrace.core.query.service.api.LiteralConstant;
+import org.hypertrace.core.query.service.api.Operator;
+import org.hypertrace.core.query.service.api.Value;
+
+class FilterToPromqlConverter {
+
+  /**
+   * only `AND` operator in filter is allowed
+   *
+   * <p>rhs of leaf filter should be literal
+   */
+  void convertFilterToString(
+      Filter filter,
+      List<String> filterList,
```

Comment:
Looks like filterList is output? nit : can this be last argument? Also can we put comments for args?

Reply:
yes, made it last argument